main.py

```diff
@@ -0,0 +1,184 @@
+from qdrant_client import QdrantClient
+from datetime import datetime, timedelta
+from fastapi.responses import JSONResponse
+from fastapi.middleware.cors import CORSMiddleware
+from fastapi.security import OAuth2PasswordRequestForm
+from fastapi import Depends, FastAPI, HTTPException, status, Path, UploadFile, File, Form
+
+# Load the configuration file
+config = utils.load_config('./config.yaml')
+
+# Load documentation of the API development and show the information when is deployed
+tag = utils.load_json(r'./data/tags.json') 
+tags = [tag[t] for t in tag]
+
+client_ia = OpenAI(api_key=config['openai']['key'])
+
+
+# Initialize Qdrant client with the host and port according to the configuration file
+client_vdb = QdrantClient(config['qdrant']['host'], port=config['qdrant']['port'])
+
+app = FastAPI(title="Machine Learning Engineer Technical Test", openapi_tags= tags)
+
+origins = ["*"]
+
+# Add CORS middleware to allow cross-origin requests
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=origins,
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+
+# Route for the home page
+@app.get("/", tags=["Home"])
+async def home():
+    """
+    Check if the API is running correctly.
+
+    Returns:
+        dict: A dictionary indicating that the API is running correctly.
+    """
+    return {'API created by Fabian Coy -2024 Machine Engineer Test': 'API to extract data using OCR, load data to QDRANT and get answers using LLM model.'}
+
+
+#### Testing to upload a file endpoint
+@app.post("/extract_data/", tags=["OCR"])
+async def extract_data(
+    file: UploadFile = File(...),
+    name: str = Form(...),
+    lastname: str = Form(...),
+    current_user: security.User = Depends(security.get_current_active_user)
+):
+    """
+    Extract data from an uploaded file using OCR.
+
+    Args:
+        file (UploadFile): The file to be processed.
+        name (str): The name of the person.
+        lastname (str): The last name of the person.
+        current_user (security.User): The current authenticated user.
+
+    Returns:
+        JSONResponse: A JSON response indicating the outcome of the data extraction process.
+    """
+    upload_folder = "uploads"
+    os.makedirs(upload_folder, exist_ok=True)
+    file_path = os.path.join(upload_folder, file.filename)
+    
+    with open(file_path, "wb") as buffer:
+        shutil.copyfileobj(file.file, buffer)
+
+    if file.filename.lower().endswith('.pdf'):
+        file_path, extracted_text = processing.handle_pdf(file_path)
+
+    elif file.filename.lower().endswith(('.png', '.jpg', '.jpeg', '.bmp', '.gif')):
+            file_path, extracted_text = processing.convert_to_png(file_path)
+    else:
+        return {"error": "Invalid file type. Please upload a correct file in format '.png', '.jpg', '.jpeg', '.bmp', '.gif' or '.pdf'"}
+
+    try:        
+        outcome = processing.process_file(file_path, name, lastname, config, extracted_text)
```

Comment:
Here you can use a DataClass to cast the outcome and check data types for the output